packages/collections/persistent/map.pony

```diff
@@ -109,6 +109,22 @@ class val HashMap[K: Any #share, V: Any #share, H: mut.HashFunction[K] val]
     end
     m
 
+  fun val add(key: K, value: val->V): HashMap[K, V, H] =>
+    """
+    This with the new (key, value) mapping.
```

Comment:
To keep this consistent with the existing package documentation I'd prefer "Return this Map with the given (key, value) mapping."